### PR TITLE
DAOS-10196 vos: Allow pool open to ignore uuid

### PR DESCRIPTION
**Pull request by ryon-jensen**
In preparation for the daos debug tool, VOS needs to be able to open
a pool without knowing the pool uuid. vos_pool_open_metrics is
updated with a parameter that indicates if the pool uuid check
should be skipped.

Signed-off-by: Ryon Jensen <ryon.jensen@intel.com>

### CODE REVIEW

File: src/include/daos_srv/vos.h [hunk 1]
```diff
@@ -294,7 +294,7 @@ vos_pool_open(const char *path, uuid_t uuid, unsigned int flags,
  */
 int
 vos_pool_open_metrics(const char *path, uuid_t uuid, unsigned int flags, void *metrics,
-		      daos_handle_t *poh);
```
Comment (jolivier23) [inline src/include/daos_srv/vos.h:297]: This reads like we are opening the pool metrics.

I would suggest we simply add an entry to pool_open_flags

VOS_POF_SKIP_UUID_CHECK

Comment (ryon-jensen) [reply src/include/daos_srv/vos.h:297]: Ya, the name of this function kind of confused me at first. I'll change to use a flag instead.

**Comment by ryon-jensen** [comment]
I updated with Jeff's feedback and did a force push since it changed a lot of this small patch.

**Review by jolivier23 — APPROVED** [review]

File: src/vos/vos_pool.c [hunk 2]
```diff
@@ -789,36 +789,39 @@ vos_pool_open_metrics(const char *path, uuid_t uuid, unsigned int flags, void *m
 	struct d_uuid		 ukey;
 	PMEMobjpool		*ph;
 	int			 rc, enabled = 1;
+	bool			 skip_uuid_check = flags & VOS_POF_SKIP_UUID_CHECK;
```
Comment (NiuYawei) [inline src/vos/vos_pool.c:792]: I don't quite see how a pool can be opened successfully without knowing pool UUID, it needs to get the SPDK blob ID from pool UUID and target ID.

What 'uuid' will be provided in this case? An empty UUID? Could you elaborate the requirement a bit?

**Review by NiuYawei — CHANGES_REQUESTED** [review]
Mark as '-1' before the concern is addressed.

File: src/vos/vos_pool.c (hunk 2 above)
Comment (ryon-jensen) [reply src/vos/vos_pool.c:792]: The primary purpose is for the use of the daos debug tool which is part of the catastrophic recovery milestone. It is meant to connect to a single vos file and be able to navigate the vos tree, inspecting and altering it. While the vos file is typically within a path that contains the pool uuid, that may not always be the case (i.e. a user sends a vos file in email for support). It was discussed in the catastrophic recovery WG meeting that being able to open a vos file while not knowing the pool uuid is desirable for the daos debug tool.

Yes, an empty UUID will be provided in this case.

File: src/vos/vos_pool.c [hunk 3]
```diff
@@ -266,7 +266,7 @@ vos_blob_unmap_cb(uint64_t off, uint64_t cnt, void *data)
 	return rc;
 }
 
-static int pool_open(PMEMobjpool *ph, struct vos_pool_df *pool_df, uuid_t uuid,
+static int pool_open(PMEMobjpool *ph, struct vos_pool_df *pool_df,
```
[the remark this answers is not in the document]
Comment (ryon-jensen) [reply src/vos/vos_pool.c:269]: I'm not sure why this is an error.

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**